overlord/snapstate/check_snap_test.go

```diff
@@ -136,12 +136,36 @@ var assumesTests = []struct {
 }, {
 	assumes: "[snapd2.15.1]",
 	version: "2.15.1",
+}, {
+	assumes: "[snapd2.15.1.2]",
+	version: "2.15.1.2",
+}, {
+	assumes: "[snapd2.15.1.2]",
+	version: "2.15.1.3",
+}, {
+	// the horror the horror!
+	assumes: "[snapd2.15.1.2.4.5.6.7.8.8]",
+	version: "2.15.1.2.4.5.6.7.8.8",
+}, {
+	assumes: "[snapd2.15.1.2.4.5.6.7.8.8]",
+	version: "2.15.1.2.4.5.6.7.8.9",
+}, {
+	assumes: "[snapd2.15.1.2]",
+	version: "2.15.1.3",
```

Comment:
If I'm reading this right, we could add a test case:
```
}, {
	assumes: "[snapd2]",
	version: "2.1",
}, {
```

Reply:
Nvm, it's just down below.

Reply:
Actually, we don't have:
```
	assumes: "[snapd2]",
	version: "3",
```
and
```
	assumes: "[snapd2]",
	version: "2",
```

Reply:
I added these test cases along with some others that we weren't covering in unit tests as well as the untested cases that the spread tests caught actually